sql/core/src/test/scala/org/apache/spark/sql/collation/CollationTypePrecedenceSuite.scala

```diff
@@ -33,6 +33,318 @@ class CollationTypePrecedenceSuite extends DatasourceV2SQLBase with AdaptiveSpar
     assert(exception.getCondition === errorClass)
   }
 
+  private def assertExplicitMismatch(df: => DataFrame): Unit =
```

Comment:
@srielau can you look at the tests to see if there is maybe something that I missed?